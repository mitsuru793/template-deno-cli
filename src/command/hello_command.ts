import { Command } from "@cliffy/command";

type Option = {
  debug?: boolean;
};

export const helloCommand = new Command()
  .name("cliffy")
  .version("0.1.0")
  .description("Command line framework for Deno")
  .option("-d, --debug", "output debug log")
  .arguments("<greetMessage:string> [targetPartson:string]")
  .action((options, greetMessage, targetPartson) => {
    main(options, greetMessage, targetPartson);
  });

function main(
  options: Option,
  greetMessage: string,
  targetPartson: string | undefined,
): void {
  const isDebug = options.debug === true;
  if (isDebug) console.log("start main function");
  console.log(`${targetPartson}, ${greetMessage}!`);
  if (isDebug) console.log("end main function");
}
