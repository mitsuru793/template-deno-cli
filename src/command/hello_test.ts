import { snapshotTest } from "@cliffy/testing";
import { helloCommand } from "./hello_command.ts";

await snapshotTest({
  name: "test stdout",
  meta: import.meta,
  async fn() {
    await helloCommand.parse(["--debug", "Hi", "Mike"]);
  },
});
