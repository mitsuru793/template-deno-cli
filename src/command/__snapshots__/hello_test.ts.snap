export const snapshot = {};

snapshot[`test stdout 1`] = `
stdout:
"start main function
Mike, Hi!
end main function
"
stderr:
""
`;
